substrafl/nodes/train_data_node.py

```diff
@@ -305,3 +315,82 @@ def summary(self) -> dict:
             "data_manager_key": self.data_manager_key,
             "data_sample_keys": self.data_sample_keys,
         }
+
+
+class SimuTrainDataNode(TrainDataNodeProtocol):
+    def __init__(self, client: substra.Client, node: TrainDataNode, strategy: ComputePlanBuilder):
+        self.organization_id = node.organization_id
+
+        self.data_manager_key = node.data_manager_key
+        self.data_sample_keys = node.data_sample_keys
+
+        self._datasamples = preload_data(
+            client=client, data_manager_key=self.data_manager_key, data_sample_keys=self.data_sample_keys
+        )
+        self._strategy = strategy
+        self._memory = SimuStatesMemory()
+
+    def init_states(self, *args, **kwargs) -> LocalStateRef:
+        return LocalStateRef(key=str(uuid.uuid4()), init=True)
+
+    def update_states(
+        self,
+        operation: RemoteDataOperation,
+        round_idx: Optional[int] = None,
+        clean_models: Optional[bool] = True,
+        *args,
+        **kwargs,
+    ) -> (LocalStateRef, Any):
+        """This function will execute the method to run on the train node with the argument
+        `_skip=True`, to execute it directly in RAM.
+
+        This function is expected to overload the `update_state` method of a TrainDataNode
+        to simulate its execution on RAM only.
+
+        Args:
+            operation (RemoteDataOperation): Object containing all the information to execute
+                the method.
+            round_idx (Optional[int]): Current round idx. Defaults to None.
+            clean_models (Optional[bool]): If set to True, the current state of the instance will
+                be saved in a SimulationIntermediateStates object. Defaults to True
+
+        Returns:
+            LocalStateRef: A bogus `LocalStateRef` to ensure compatibility with the execution of the
+                experiment.
+            Any: the output of the execution of the method stored in `operation`.
+        """
+        method_name = operation.remote_struct._method_name
+        method_parameters = operation.remote_struct._method_parameters
+
+        method_parameters["shared_state"] = operation.shared_state
+        method_parameters["datasamples"] = self._datasamples
+
+        # To be compatible with custom compute plans
```

Comment:
I would expand this comment to be more explicit for people who do not have in mind the intricacies of Strategy vs ComputeGraph implementations